### PR TITLE
Run e2e tests inside a docker container

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Related to and blocks #1017 by @sarayourfriend 

I originally started working on this directly in the same PR to address #1017 but decided to split this work out and apply it to the existing e2e tests to minimize the scope of the visual regression testing PR.

### Note about the line count for this PR!

The vast majority of those changes are in the updated tapes due to the sudden introduction of Brotli compression halfway through developing this. The main places to focus reviews on are the following:

1. `playwright.config.ts`
2. `docker-compose.playwright.ts`
3. `proxy.js`

### Rationale

In order for visual regression tests to run in docker (to alleviate cross-platform rendering differences causing false-positives in the snapshot tests) we need to be able to have the full playwright stack working in docker due to networking complications.

Because our app is SSR and client side rendered, there are two contexts in which API requests are made:

1. In the SSR application
4. In the docker contained headless browser that is being controlled by playwright

If we just run playwright inside of docker, then we run into an issue where our API url has to be different for the client and the server, because the server will be on the host network and can use `localhost`, whereas the client will be in the docker network and will need to use something like `host.docker.internal`. This complicates the code and also makes debugging difficult.

### Solution

To fix this, we run everything inside a single Playwright docker container so that they all share the same network _and_ (crucially) can communicate over plain old `localhost`. Without being able to use `localhost` (i.e., relying on docker-compose networking) we can lots of weird SSL errors with the browser automatically redirecting requests to places like `http://web` or `http://api` to `https`. These fail because we don't have SSL set up locally.

Instead, I discovered that Playwright actually has built into it the ability to run and wait for a server to start! This means we can run the talkback proxy _and_ the Nuxt server all inside the same container as Playwright, still get the benefits of the Playwright container's browser, all without the hassle of setting up local SSL for everyone trying to run the e2e tests locally.

The only snag here is that writing new tapes from talkback needs to have some permissions juggling fixed for Linux. Apparently this is a non-issue on macOS and Windows because the volumes are shared in a different way, but on Linux files written by docker containers into the host will all have root permissions. To fix this, we can just run `chmod` on them after updating the tapes. To prevent having to run a `sudo` script inside our package.json, I chose to just use the `node:alpine` image to run `chmod` as the image will already be present on your machine if you run the e2e tests because it's used as the base image for the running Nuxt app. I think we could also use the Playwright image but we need to interpolate the Playwright version to avoid downloading extraneous versions and `node:alpine` doens't have that problem.

I had to use `npm-run-all` because pre/post scripts in npm do not run if the main script fails, so if you try to update tapes and the tests don't pass, then the permissions fix won't run. `npm-run-all` with the `--continue-on-error` flag set will run all the scripts in sequence even if one of them fails while also still producing a non-zero exit code if any of the scripts produce non-zero exit codes.

### Meta commentary about the PR

There are a few long caveat laden comments that needed to be added to explain everything that was going on, specifically with a few edge cases here and there. If when reviewing you find any other places that need clarification, don't hesitate to ask me to add more comments. Given the complexity and depth of this change I think more explanatory comments are best, for our own sanity's sake down the line when we're trying to understand how this all works again.

I actually think that the solution I landed on is ultimately simpler than the solutions I explored along the way. For example, this still uses `docker-compose` to run the Playwright container, but that's mostly just so that we don't have to stuff all the options into a long `docker run` line.

### Related fixes

Because I was debugging all sorts of broken https requests, I was carefully inspecting the tapes and requests being made by the frontend. I discovered that the thumbnail routes were not being proxied. This PR also fixes that (hence the enormous number of new tapes being added).

Halfway through working on this I ran into an issue where some responses that were being recorded by talkback when updating/creating new tapes were compressed. Talkback, for some reason, doesn't decompress them for you. That's fine because `zlib` is built into Node anyway and we can easily decompress and recompress responses. I don't know why we didn't run into this previously but as I mentioned in the comment in `proxy.js` I think it's actually good that we're dealing with it because it more closely mirrors the situation in production.

It does, however, raise the question of why some responses _aren't_ Brotli compressed and others are.

In light of @obulat's existing PR to fix up some of the tape generation logic in https://github.com/WordPress/openverse-frontend/pull/1045 we can hold off on merging this PR until her changes are in. I think it'll be easier to merge in her changes to this PR than the other way around. @obulat feel free to correct me if you think the other way will be easier! I'm not sure if your PR is running into the same Brotli compression issue I ran into with this one, but if so then the other way around (merge this first then rebase the other branch) might be easier. I will leave that up to you though, Olga.

## Testing Instructions

Checkout this PR and run `pnpm test:e2e` and `pnpm test:e2e:update-tapes` and ensure that they work (i.e., that talkback responses are being used/created, file permissions are set correctly, etc). Also try `pnpm test:e2e:debug` with and without a local Nuxt server running. Both should work. `pnpm generate-playwright-tests` is unaffected, but I renamed it in anticipation of visual-regression being added and the generate script no longer being specific to just "e2e" tests.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
I've gotten _very_ far with this. I think the final issue that remains is to fix the various SSL errors that happen. You can see these by modifying the `playwright.config.js` to include `use: { trace: 'on' }`, extracting the `trace.zip` created in `test-results` and opening the `trace.trace` file. Search for `https` and you'll find several requests to both `https://api:3000` and `https://web:8443`. I'm not sure what is causing this to be honest but I ran into it with the visual regression PR I was working on as well.

File: test/e2e/translation-banner.spec.js [hunk 1]
```diff
@@ -21,9 +21,7 @@ test('Can close the translation banner', async ({ page }) => {
   await page.goto('/ru/search/')
   await page.click('[aria-label="Закрыть"]')
 
-  const banner = await page.locator(
-    '.span:has-text("Help us get to 100 percent")'
-  )
+  const banner = page.locator('.span:has-text("Help us get to 100 percent")')
```
Comment (sarayourfriend) [inline test/e2e/translation-banner.spec.js:24]: `page.locator` doesn't return a `Promise`. Completely unrelated to this PR but it was bugging me while I was debugging this particular test :sweat_smile:

**Comment by sarayourfriend** [comment]
This PR is so close. Now just dealing with file permissions issues.

Locally I can't reproduce this. If I change the playwright container to not run using the `pwuser` user I get different errors locally about `npm` not being able to access some log ifles:

```
openverse-frontend-playwright-1  | [WebServer] glob error [Error: EACCES: permission denied, scandir '/root/.npm/_logs'] {
openverse-frontend-playwright-1  | [WebServer]   errno: -13,
openverse-frontend-playwright-1  | [WebServer]   code: 'EACCES',
openverse-frontend-playwright-1  | [WebServer]   syscall: 'scandir',
openverse-frontend-playwright-1  | [WebServer]   path: '/root/.npm/_logs'
openverse-frontend-playwright-1  | [WebServer] }
```

I'll dig into this Monday. For now I'm just glad to have gotten this far and have everything working and running properly locally!

**Comment by sarayourfriend** [comment]
In the end it seemed like there was no viable solution for running the scripts with `npm` due to `npm`'s rather obstinate log placement situation.

Turns out it was trivial to just build our own image based on the playwright one we needed and install `pnpm` in there to allow us to run everything with `pnpm`. So then even though we still need to run everything in the container as root (in order to allow the container to write the build to the action workspace, though why this problem doesn't present locally on my machine is beyond me) we don't run into the issues with `npm` inside the container trying to write logs to a directory outside the mounted volume scope (the action runner's `/root` directory). `npm` does not allow configuring the log output location so it's basically a moot point whether or not to use it.

Building our own image that installs `pnpm` does add some extra time to the final `e2e` run but it's negligible compared to the time spent building the Nuxt app inside the playwright container. If we wanted to cut that time (or rather amortize it) we could publish an image to our GHCR whenever we update the Playwright version in the `pnpm-lock.yaml`. It could be a nice way to not have to pull the playwright version from `pnpm ls` to interpolate it locally or in the CI but, again, it's honestly a negligible amount of time that is spent building that base image.

File: test/e2e/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import type { PlaywrightTestConfig } from '@playwright/test'
+
+const config: PlaywrightTestConfig = {
+  testDir: '.',
+  webServer: {
+    /**
+     * Note: You can speed up local testing by switching `prod` out for `start`
```
Comment (sarayourfriend) [inline test/e2e/playwright.config.ts:7]: It actually occurred to me sometime last night that we could add like a `slow-start` script that sleeps for a second or two before actually starting the nuxt server.

Or we could combine it with the [`wait-for` script that I was using](https://github.com/Eficode/wait-for) before I discovered that Playwright had this `webserver` option built into it. `start` could see if `API_URL` was pointing to localhost and then `wait-for $API_URL -- pnpx nuxt start` or something.

File: test/proxy.js [hunk 3]
```diff
@@ -9,18 +9,79 @@
...
+ *
+ * A note to future contributors: If you find that JSON.parse is
+ * complaining about unknown characters then you're either
+ * dealing with an error response that isn't being caught by
+ * the status check on the first line OR you've discovered
+ * another compression algorithm being used (or, of course, it
+ * could be something else entirely). Don't discount that it could
+ * be something else, but I'd check those two things first
+ * before digging elsewhere.
+ *
```
Comment (zackkrida) [inline test/proxy.js:59]: Your approach is likely better and more robust, anyway, but I believe we could also bypass cache hits entirely by adding a random query param to the URL.

Comment (sarayourfriend) [reply test/proxy.js:59]: Ah we could! Good to keep in mind for other types of testing. But I do think maintaining an API proxy that mirrors production as close as possible is beneficial. Normal requests aren't sending random query params after all.

File: test/e2e/playwright.config.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+     * ConsoleMessage strings that refer to incorrect responses. Once you've
+     * verified that there is indeed a tape saved for the response in question,
+     * switch this back to `prod` and see if your tests pass.
+     */
+    command: './node_modules/.bin/npm-run-all -p -r talkback prod',
+    timeout: 120000,
+    port: 8443,
+    reuseExistingServer: !process.env.CI || process.env.PWDEBUG === '1',
+    env: {
+      API_URL: 'http://localhost:3000',
```
Comment (obulat) [inline test/e2e/playwright.config.ts:26]: Should we change this port value to `49152` as in #1045, or does it not matter when inside Docker?

**Review by obulat — COMMENTED** [review]
It would definitely be easier for me to merge #1045 before this PR. I don't know if that one needs Brotli changes because it didn't really cause problems, so I guess it's safe to merge without them.

I am not able to run `pnpm run test:e2e` due to `npm-run-all` not being available: 
```
=> [2/2] RUN npm install -g pnpm                                          2.1s
 => exporting to image                                                     0.1s
 => => exporting layers                                                    0.1s
 => => writing image sha256:86b53a2d6208cc6b947777f30c5902c51de98dad664e9  0.0s
 => => naming to docker.io/library/openverse-frontend_playwright           0.0s
[+] Running 2/2
 ⠿ Network openverse-frontend_default         Created                      0.0s
 ⠿ Container openverse-frontend-playwright-1  Created                      0.1s
Attaching to openverse-frontend-playwright-1
openverse-frontend-playwright-1  |
openverse-frontend-playwright-1  | > openverse-frontend@2.2.1 test:e2e:local /app
openverse-frontend-playwright-1  | > playwright test -c test/e2e
openverse-frontend-playwright-1  |
openverse-frontend-playwright-1  | [WebServer] /bin/sh: 1: ./node_modules/.bin/npm-run-all: not found
openverse-frontend-playwright-1  | Error: Process from config.webServer was not able to start. Exit code: 127
```

File: test/e2e/playwright.config.ts (hunk 4 above)
Comment (sarayourfriend) [reply test/e2e/playwright.config.ts:26]: Oh, I had a comment about changing this once you PR is merged.

It needs to match whatever talkback is using so once your PR is merged, if it goes before this one, we need to update it to the new port number you chose.

**Comment by sarayourfriend** [comment]
@obulat you need to run `pnpm install` before running `pnpm test:e2e`. We could add it as a pre-script for `test:e2e` like we do for `dev` if that would be helpful.

**Comment by obulat** [comment]
> @obulat you need to run `pnpm install` before running `pnpm test:e2e`. We could add it as a pre-script for `test:e2e` like we do for `dev` if that would be helpful.

Oh no! I was just following the testing instructions, and didn't realize that there was a new package that should be installed locally :) I guess it would be nice to add `pnpm install` pre-script.

I couldn't run the tests in the beginning, and had to increase the timeout. Then, the tests run well, and at the end I can see a piece of an error like this:
```
 ⠿ Container openverse-frontend-playwright-1  Stopped                                                                                                                                                                                 0.0s
ERRO[0209] 0      
```
I guess it is caused by `--abort-on-container-exit`, and that's why we use the exit code from playwright, right? `--exit-code-from playwright`

File: bin/playwright.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+
+# todo: find a way to make this portable for Windows users so it can be used for the package.json script
```
Comment (obulat) [inline bin/playwright.sh:3]: I wonder if we can re-write this script using Node/JS  to make it cross-platform? Not for this PR :)

File: package.json [hunk 6]
```diff
@@ -14,22 +14,27 @@
...
     "test": "jest",
-    "test:all": "pnpm e2e:ci & jest",
+    "test:all": "pnpm test:e2e & jest",
     "test:watch": "jest --collectCoverage=false --watch",
-    "test:e2e": "playwright test",
-    "test:e2e-debug": "PWDEBUG=1 playwright test",
-    "build-and-e2e": "start-server-and-test 'cross-env API_URL=http://localhost:3000 nuxt build && pnpm start' http://localhost:8443 test:e2e",
+    "talkback": "node ./test/proxy.js",
+    "test:e2e": "cross-env PLAYWRIGHT_VERSION=1.17.1 TEST_TYPE=e2e docker-compose -f docker-compose.playwright.yml up --abort-on-container-exit --exit-code-from playwright",
+    "test:e2e:local": "playwright test -c test/e2e",
```
Comment (obulat) [inline package.json:27]: I didn't know we can have double `:` in the script name 👍

File: TESTING_GUIDELINES.md [hunk 7]
```diff
@@ -92,36 +92,46 @@ Openverse uses [Vue Testing Library](https://testing-library.com/docs/vue-testin
...
-Before running the e2e tests, install the browsers that Playwright needs:
+Our end-to-end test suite runs inside a Playwright docker container in order to prevent cross-platform browser differences from creating flaky test behavior.
 
-```
-pnpx install playwright
-```
+Having docker and docker-compose is a pre-requisite to running the end-to-end tests locally. Please follow [the relevant instructions for your operating system for how to install docker and docker-compose](https://docs.docker.com/get-docker/). If you're on Windows 10 Home Edition, please note that you'll need to [install and run docker inside WSL2](https://www.freecodecamp.org/news/how-to-run-docker-on-windows-10-home-edition/).
 
-If you don't have the app running, start by running it in the dev mode:
+If it's not possible for you to run docker locally, don't fret! Our CI will run it on every pull request and another contributor who is able to run the tests locally can help you develop new or update existing tests for your changes.
```
Comment (obulat) [inline TESTING_GUIDELINES.md:101]: Nice addition!

**Review by obulat — APPROVED** [review]

**Comment by sarayourfriend** [comment]
> I guess it is caused by --abort-on-container-exit, and that's why we use the exit code from playwright, right? --exit-code-from playwright

I think this can actually be removed now as we don't have multiple containers :thinking: I'll see if it's possible to remove it. The `ERRO[0209] 0` output is confusing. The last number is the exit code. I'm assuming the number inside the brackets is a PID or something. It's just confusing that it always uses `ERRO` at the start even if the process exited with a non-zero exit code :roll_eyes:

File: bin/playwright.sh (hunk 5 above)
Comment (sarayourfriend) [reply bin/playwright.sh:3]: We could... node's sub-process control is pretty weird, I could never really figure out how best to manage them.

File: package.json (hunk 6 above)
Comment (sarayourfriend) [reply package.json:28]: Yeah! I've used [this package (`scripty`)](https://www.npmjs.com/package/scripty) in the past when I worked on the [debtcollective/dispute-tools](https://github.com/debtcollective/dispute-tools) and it uses `:` as the path delimiter. It lets you define separate scripts for windows and unix-types and reference them by the same name. Very helpful! If we don't switch to `just` we could use `scripty` to more easily create cross-platform scripts for the frontend.

**Comment by sarayourfriend** [comment]
@obulat I tried to use `docker-compose ... run playwright` but ran into all sorts of weird issues with nuxt's build lock file not getting cleared for some reason! Switched back to `up` and it works perfectly :shrug:

File: src/utils/env.ts [hunk 8]
```diff
@@ -1,9 +1,11 @@
+const apiUrl = process.env.API_URL ?? 'https://api.openverse.engineering/'
+
 /**
  * Default environment variables are set on this key. Defaults are fallbacks to existing env vars.
  * All boolean values should be designed to be false by default.
  */
 export const env = {
-  apiUrl: process.env.API_URL ?? 'https://api.openverse.engineering/',
+  apiUrl: apiUrl.endsWith('/') ? apiUrl : `${apiUrl}/`,
```
Comment (zackkrida) [inline src/utils/env.ts:8]: 👍

**Comment by zackkrida** [comment]
Getting Brotli related errors in my attempts to test this:

<details>
<summary>Full terminal output</summary>

```bash
front on  try/e2e-docker +/- [$] via  v16.13.0
❯ pnpm test:e2e

> openverse-frontend@2.2.1 pretest:e2e /Users/zackkrida/Code/openverse/front
> pnpm install

Lockfile is up-to-date, resolution step is skipped
Already up-to-date

> openverse-frontend@2.2.1 prepare /Users/zackkrida/Code/openverse/front
> husky install

husky - Git hooks installed

> openverse-frontend@2.2.1 test:e2e /Users/zackkrida/Code/openverse/front
> cross-env PLAYWRIGHT_VERSION=1.17.1 TEST_TYPE=e2e docker-compose -f docker-compose.playwright.yml up --abort-on-container-exit --exit-code-from playwright

WARN[0000] Found orphan containers ([front_web_1]) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
[+] Running 1/0
 ⠿ Container front-playwright-1  Created                                                                                                                                                                                                                                                                                 0.0s
Attaching to front-playwright-1
front-playwright-1  |
front-playwright-1  | > openverse-frontend@2.2.1 test:e2e:local /app
front-playwright-1  | > playwright test -c test/e2e
front-playwright-1  |
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.272Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-106.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.280Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-107.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.397Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-114.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.471Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-117.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.595Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-121.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.708Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-125.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:43.962Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-13.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:44.673Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-15.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.234Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-175.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.253Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-179.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.297Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-180.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.311Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-181.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.413Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-185.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.428Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-186.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.439Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-187.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.623Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-190.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.723Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-21.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:45.759Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-24.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:46.021Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-30.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:46.470Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-42.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:47.053Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-58.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:47.056Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-59.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:47.074Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-60.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:47.089Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-61.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:47.340Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-68.json5 Unsupported content-encoding br
front-playwright-1  | [WebServer] 2022-03-14T14:37:47.370Z [Openverse e2e proxy] [ERROR] Error reading tape test/tapes/response-70.json5 Unsupported content-encoding br
```

</details>

**Comment by sarayourfriend** [comment]
@zackkrida that's a talkback error: https://github.com/sarayourfriend/talkback/blob/7226f9c24ad78957575750cc39a3b17787270602/src/utils/content-encoding.ts#L39

Just need to update to the most recent version of talkback since my PR to add Brotli support was merged upstream: https://github.com/ijpiantanida/talkback/releases/tag/v3.0.1

**Comment by sarayourfriend** [comment]
@zackkrida Ready for re-review.

**Comment by sarayourfriend** [comment]
@zackkrida I'm going to push another commit to solve an issue with duplicate tapes that @obulat pointed out to me. Should be pushing it within the next hour or so.

**Comment by sarayourfriend** [comment]
@obulat What command did you use to run the debug? `pnpm test:e2e:debug`? It looks like somehow the tests were run without a `TEST_TYPE` for one of the docker-based approaches... :thinking:

**Comment by sarayourfriend** [comment]
> but only if I increase the timeout in playwright.config.ts

I'll increase the timeout to five minutes I think.

**Comment by sarayourfriend** [comment]
@obulat I tried running `pnpm test:e2e:debug` locally and it worked fine for me :confused: Unfortunately we can't reuse the docker approach for debugging tests because docker would need to connect to a running Xorg server to be able to display the headed browsers.... and configuring that just for _one_ OS is hard enough, making it cross platform to macOS, Linux, and Windows (even with WSL) would require a lot of work. Maybe one day we can tackle that but for now `test:e2e:debug`'s approach is the only way :disappointed:

**Review by obulat — APPROVED** [review]
Well, debugging works locally, and e2e tests run well in docker. This is such a great improvement, and it's great that you split this part of the change into a separate smaller PR:)

**Comment by sarayourfriend** [comment]
@zackkrida can you re-review today if you have time? I'd like to merge this to avoid any more merge conflicts. Thanks!

**Comment by sarayourfriend** [comment]
@krysal Would you be able to review this when you have a chance?

**Review by zackkrida — APPROVED** [review]
LGTM. This is working flawlessly for me now. My only nit is that after running `pnpm test:e2e`, this is the only output for 2-3 minutes:

```shell
Starting front_playwright_1 ... done
Attaching to front_playwright_1
playwright_1  |
playwright_1  | > openverse-frontend@2.2.1 test:e2e:local /app
playwright_1  | > playwright test -c test/e2e
playwright_1  |
playwright_1  |
```

If we could log a message or something right away that indicates that a bunch of background work is happening, that might reassure users while this is loading. I worry about folks with slower machines than my m1 macbook pro. I left an inline comment with a suggestion on how to do this.

File: docker-compose.playwright.yml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    build:
+      context: .
+      dockerfile: Dockerfile.playwright
+      args:
+        - PLAYWRIGHT_VERSION=${PLAYWRIGHT_VERSION}
+    volumes:
+      - .:/app
+    user: ${PW_USER:-pwuser}
+    working_dir: /app
+    command: pnpm test:${TEST_TYPE}:local
```
Comment (zackkrida) [inline docker-compose.playwright.yml:13]: Could do something here to log a message like so:

```
   command: >
      sh -c "echo Some type of message here
             pnpm test:${TEST_TYPE}:local"
```

**Pull request merged**